src/main/java/htsjdk/samtools/cram/structure/Container.java

```diff
@@ -59,10 +63,57 @@
      */
     public long offset;
 
+    /**
+     * Set this container's sequence ID and derive its {@link SliceType} from that value
+     * @param sequenceId the sequence ID
+     */
+    public Container(final int sequenceId) {
+        this.sequenceId = sequenceId;
+        this.containerType = SliceType.bySequenceId(sequenceId);
+    }
+
+    public static Container newMultiRefContainer() {
```

Comment:
As long as we continue to have ` ContainerFactory` class I wonder if these should live there.

Reply:
Hm, interesting.  My inclination is to remove ContainerFactory in a future Container iteration.

Reply:
Agreed - I just thought in the meantime it was suboptimal to not have them all together, but its not that important if its temporary.

Reply:
See my comment below - I think it would eliminate most of the pass-through method if we just make the one concrete implementation of `ReferenceContext` be the currency used everywhere, but if not let me know and we can revisit this.